postgres/datadog_checks/postgres/statements.py

```diff
@@ -61,7 +63,46 @@
     'query': 'query',
 }
 
-DEFAULT_STATEMENT_METRIC_LIMITS = {k: (10000, 10000) for k in PG_STAT_STATEMENTS_METRIC_COLUMNS.keys()}
+# These limits define the top K and bottom K unique query rows for each metric. For each check run the
+# max metrics sent will be sum of all numbers below (in practice, much less due to overlap in rows).
+DEFAULT_STATEMENT_METRIC_LIMITS = {
+    'calls': (400, 0),
+    'total_time': (400, 0),
+    'rows': (400, 0),
+    'shared_blks_hit': (50, 0),
+    'shared_blks_read': (50, 0),
+    'shared_blks_dirtied': (50, 0),
+    'shared_blks_written': (50, 0),
+    'local_blks_hit': (50, 0),
+    'local_blks_read': (50, 0),
+    'local_blks_dirtied': (50, 0),
+    'local_blks_written': (50, 0),
+    'temp_blks_read': (50, 0),
+    'temp_blks_written': (50, 0),
+    # Synthetic column limits
+    'avg_time': (400, 0),
+    'shared_blks_ratio': (0, 50),
+}
+
+
+def generate_synthetic_rows(rows):
+    """
+    Given a list of rows, generate a new list of rows with "synthetic" column values derived from
+    the existing row values.
+    """
+    synthetic_rows = []
+    for row in rows:
+        new = copy.copy(row)
```

Comment:
Do we need a stronger `deepcopy`? Depends on what `row` contains. If it's just a tuple of scalars (strings, numbers, …) then please ignore.

```suggestion
        new = copy.deepcopy(row)
```

Reply:
Shallow copy is fine here as there will never be nested objects in rows.